File: lab#1/java/Student.java
import java.lang.String;
import java.util.ArrayList;

public class Student {
    private String nume;
    private String prenume;
    private long IDNP;
    private int promotia;
    private String grupa;
    private int notaAM;
    private int notaPC;
    private int notaA;
    private int notaBI;
    private int notaE;
    private double mediaGenerala;
    private int credite;

    void Student(String nume, String prenume, long IDNP, int promotia, String grupa, int notaAM, int notaPC, int notaA, int notaBI, int notaE){
        this.nume = nume;
        this.prenume = prenume;
        this.IDNP = IDNP;
        this.promotia = promotia;
        this.grupa = grupa;
        this.notaAM = notaAM;
        this.notaPC = notaPC;
        this.notaA = notaA;
        this.notaBI = notaBI;
        this.notaE = notaE;
        this.mediaGenerala = (notaAM + notaPC + notaA + notaBI + notaE) / 5.0;
        int ects = 0;
        if(notaAM>4) { ects+=6;}
        if(notaPC>4) { ects+=6;}
        if(notaA>4) { ects+=6;}
        if(notaBI>4) { ects+=6;}
        if(notaE>4) { ects+=6;}
        this.credite = ects;
    }


    private static Student read() {
        Console c = System.console();
        String nume = c.readLine("Numele:");
        String prenume = c.readLine("Prenumele:");
        long IDNP = c.readLine("IDNP:");
        int promotia = c.readLine("promotia:");
        String grupa = c.readLine("grupa:");
        int notaAM = c.readLine("nota Analiza Matematica:");
        int notaPC = c.readLine("nota Programarea C:");
        int notaA = c.readLine("nota Algoritmica:");
        int notaBI = c.readLine("nota Bazele Informaticii:");
        int notaE = c.readLine("nota Engleza:");
        Student student = new Student(nume,prenume,IDNP,promotia,grupa,notaAM,notaPC,notaA,notaBI,notaE);
        return student;
    }

    public static void main(String[] args) {
        ArrayList<Student> arrayList = new ArrayList<Student>();
        arrayList.add(read());
        System.out.println("hello world");
        System.out.println(arrayList.size());
    }
}

File: lab#1/cpp/student.cpp
#include<iostream>
#include<string>
#include<fstream>
#include<vector>
using namespace std;
struct student {
    string nume;
    string prenume;
    long IDNP;
    int promotia;
    string grupa;
    int notaAM;
    int notaPC;
    int notaA;
    int notaBI;
    int notaE;
    double mediaGenerala;
    int credite;
} stud;
int main(){
    ofstream myfile;
    char c;
    long id;
    int pos=0;
    std::vector< student > students;
    while(true) {
        cout<<"introdu datele"<<endl;
        cout<<"numele:";
        cin>>stud.nume;    
        cout<<"prenumele:";
        cin>>stud.prenume;
        cout<<"IDNP:";
        cin>>stud.IDNP;
        cout<<"promotia:";
        cin>>stud.promotia;
        cout<<"grupa:";
        cin>>stud.grupa;
        cout<<"Nota Analiza Matematica:";
        cin>>stud.notaAM;
        cout<<"Nota Programarea C:";
        cin>>stud.notaPC;
        cout<<"Nota Algoritmica:";
        cin>>stud.notaA;
        cout<<"Nota Bazele Informaticii:";
        cin>>stud.notaBI;
        cout<<"Nota Engleza:";
        cin>>stud.notaE;
        stud.mediaGenerala=(stud.notaAM+stud.notaPC+stud.notaA+stud.notaBI+stud.notaE)/5.0;
        stud.credite=0;
        if(stud.notaAM>4) { stud.credite+=6;}
        if(stud.notaPC>4) { stud.credite+=6;}
        if(stud.notaA>4) { stud.credite+=6;}
        if(stud.notaBI>4) { stud.credite+=6;}
        if(stud.notaE>4) { stud.credite+=6;}
        students.push_back(stud);
        cout<<"enter another student?[y/n]"<<endl;
        cin>>c;
        if(c=='n') break;
    }
    cout<<"cauta student(dupa ID):";
    cin>>id;
    
    for(student stud : students) {
        if(stud.IDNP==id) {
            cout<<"found:"<<endl;
            cout<<stud.nume<<" "<<stud.prenume<<" "<<stud.mediaGenerala<<" "<<stud.credite<<endl;
        }
    }

    cout<<"numarul de studenti:"<<students.size()<<endl;
    cout<<"sterge student(dupa ID):";
    cin>>id;
    pos=-1;
    for(student stud : students) {
        if(stud.IDNP==id) {
            pos+=1;
            break;
        }
        pos+=1;
    }
    students.erase(students.begin()+pos);
    cout<<"numarul de studenti:"<<students.size()<<endl;

    myfile.open("students.txt");
    for(student stud: students) {
        myfile<<stud.nume<<" "<<stud.prenume<<" "<<stud.mediaGenerala<<" "<<stud.credite<<endl;
    }
    myfile.close();
    return 0;
    
}


